src/System.Net.Security/src/System/Net/_SecureChannel.cs

```diff
@@ -0,0 +1,1487 @@
+        //   HeaderSize             - Header & trailer sizes used in the TLS stream
+        //   TrailerSize -
+        //
+        internal X509Certificate LocalServerCertificate
+        {
+            get
+            {
+                return _serverCertificate;
+            }
+        }
+
+        internal X509Certificate LocalClientCertificate
+        {
+            get
+            {
+                return _selectedClientCertificate;
+            }
+        }
+
+        internal bool IsRemoteCertificateAvailable
+        {
+            get
+            {
+                return _isRemoteCertificateAvailable;
+            }
+        }
+
+        private unsafe static class UnmanagedCertificateContext
+        {
+            internal static X509Certificate2Collection GetStore(SafeFreeCertContext certContext)
+            {
+                X509Certificate2Collection result = new X509Certificate2Collection();
+
+                if (certContext.IsInvalid)
+                    return result;
+
+                Interop.Crypt32.CERT_CONTEXT context =
+                    Marshal.PtrToStructure<Interop.Crypt32.CERT_CONTEXT>(certContext.DangerousGetHandle());
+
+                if (context.hCertStore != IntPtr.Zero)
+                {
+                    X509Store store = null;
+                    try
+                    {
+                        store = X509StoreExtensions.CreateFromNativeHandle(context.hCertStore);
+                        result = store.Certificates;
+                    }
+                    finally
+                    {
+                        if (store != null)
+                            store.Dispose();
+                    }
+                }
+                return result;
+            }
+        }
+
+        //
+        // Extracts a remote certificate upon request.
+        //
+        internal X509Certificate2 GetRemoteCertificate(out X509Certificate2Collection remoteCertificateStore)
+        {
+            remoteCertificateStore = null;
+
+            if (_securityContext == null)
+                return null;
+
+            GlobalLog.Enter("SecureChannel#" + Logging.HashString(this) + "::RemoteCertificate{get;}");
+            X509Certificate2 result = null;
+            SafeFreeCertContext remoteContext = null;
+            try
+            {
+                remoteContext = SSPIWrapper.QueryContextAttributes(GlobalSSPI.SSPISecureChannel, _securityContext, Interop.Secur32.ContextAttribute.RemoteCertificate) as SafeFreeCertContext;
+                if (remoteContext != null && !remoteContext.IsInvalid)
+                {
+                    result = new X509Certificate2(remoteContext.DangerousGetHandle());
+                }
+            }
+            finally
+            {
+                if (remoteContext != null)
+                {
+                    remoteCertificateStore = UnmanagedCertificateContext.GetStore(remoteContext);
```

Comment:
This was odd to me too: GetStore is actually returning an `X509Certificate2Collection` in the original code. I've renamed `GetStore` to: `UnmanagedCertificateContext.GetRemoteCertificatesFromStoreContext(remoteContext);`

This is the place where we need to extract certs from `CERT_CONTEXT.hCertStore` but `X509Store(IntPtr)` is missing from CoreFX.